src/Stratis.Bitcoin.IntegrationTests/Mempool/MemoryPoolTests.cs

```diff
@@ -184,31 +184,34 @@ public void TxMempoolBlockDoublespend()
                 }
 
                 // Test 1: block with both of those transactions should be rejected.
-                Block block = stratisNodeSync.GenerateBlockManually(spends);
+                var tipBeforeBlockCreation = stratisNodeSync.FullNode.Chain.Tip;
+                Assert.Throws<ConsensusException>(() => { Block block = stratisNodeSync.GenerateBlockManually(spends); });
```

Comment:
Did this not throw before?

Reply:
@bokobza yes prior to activation, we didn't let consensus exceptions bubble to PowMining etc. So now we have to catch it and ensure that the tip didnt advance cause an invalid block was pushed to consensus.

Reply:
makes sense